axis/global_parameters.f95

```diff
@@ -0,0 +1,147 @@
+real(8),dimension(0:9)::m_pl=(/ 0d0,&
+    1.6601367952719304D-07 , 2.4478383396645447D-06, 3.0404326462685257D-06,&
+    3.2271514450538743D-07 , 9.547919384243222D-04 , 2.858859806661029D-04 ,&
+    4.3662440433515637D-05 , 5.151389020535497D-05 , 7.407407407407407D-09 /)
+! Used: http://ssd.jpl.nasa.gov/?planet_phys_par
+
+!----------------------------------------------------------------------------------------------
+integer,parameter:: gp_max_order = 20
+
+!----------------------------------------------------------------------------------------------
+type idmrow_2body
+    integer,dimension(4)::resonance
+    real(8):: res_a
+end type idmrow_2body
+
+type idmrow_3body
+    integer,dimension(6)::resonance
+    real(8):: res_a
+end type idmrow_3body
+
+type idm_2body
+    type(idmrow_2body),dimension(:),allocatable:: matrix
+end type idm_2body
+
+type idm_3body
+    type(idmrow_3body),dimension(:),allocatable:: matrix
+end type idm_3body
+
+type(idm_2body),dimension(9):: idmatrix_2body
+type(idm_3body),dimension(72):: idmatrix_3body
+!----------------------------------------------------------------------------------------------
+contains
+
+!----------------------------------------------------------------------------------------------
+integer function planet_id(s)
+! Get planet ID
+! Given:
+!   s - planet name IN CAPITALS
+! Returns:
+!   <integer> - ID
+    character(*) s
+    integer i
+
+    select case (s)
+        case('MERCURY')
+            i=1
+        case('VENUS')
+            i=2
+        case('EARTHMOO')
+            i=3
+        case('MARS')
+            i=4
+        case('JUPITER')
+            i=5
+        case('SATURN')
+            i=6
+        case('URANUS')
+            i=7
+        case('NEPTUNE')
+            i=8
+        case('PLUTO')
+            i=9
+        case default
+            i=0
+            write(*,*)'Warning! Detected incorrect planet name, be careful.'
+    end select
+    planet_id=i
+end function planet_id
+
+!----------------------------------------------------------------------------------------------
+function planet_name(i) result(s)
+! Get planet ID
+! Given:
+!   s - planet name IN CAPITALS
+! Returns:
+!   <integer> - ID
+    character(8) s
+    integer i
+
+     select case (i)
+         case(1)
+             s='MERCURY'
+         case(2)
+             s='VENUS'
+         case(3)
+             s='EARTHMOO'
+         case(4)
+             s='MARS'
+         case(5)
+             s='JUPITER'
+         case(6)
+             s='SATURN'
+         case(7)
+             s='URANUS'
+         case(8)
+             s='NEPTUNE'
+         case(9)
+             s='PLUTO'
+         case default
+             write(*,*)'Warning! Detected incorrect planet id, be careful.'
+     end select
+ end function planet_name
+
+!----------------------------------------------------------------------------------------------
+integer function nod(a,b)
```

Comment:
Refreshed the function, named gcd, deleted 1 variable and modified the algorithm.